app/src/main-process/menu/build-default-menu.ts

```diff
@@ -369,7 +376,9 @@ export function buildDefaultMenu({
         click: emit('discard-all-changes'),
       },
       {
-        label: __DARWIN__ ? 'Stash All Changes…' : '&Stash all changes…',
+        label: askForConfirmationWhenStashingAllChanges
+          ? confirmStashAllChangesLabel
+          : stashAllChangesLabel,
```

Comment:
I noticed that we're not updating the label of the app menu after it's initially set.

It would be cool to fix but I also don't think it's a big deal (specially since I don't see that we update labels for any other app menu items in `menu-update.ts`).

Anyways, I think that we decide to not update the label here we should have a fixed label (e.g the one with ellipsis) so it remains always static and it does not vary depending on the initial state when Desktop is launched

Reply:
> I noticed that we're not updating the label of the app menu after it's initially set.

That's odd, it's updating for me. Are you sure you rebuilt the main process before testing?

![Screen Shot 2020-06-18 at 14 05 53](https://user-images.githubusercontent.com/634063/85018108-eda7c180-b16c-11ea-8a4c-0a69802b64a3.png)
![Screen Shot 2020-06-18 at 14 06 09](https://user-images.githubusercontent.com/634063/85018114-f0a2b200-b16c-11ea-9b7d-b3d69326733a.png)

Reply:
Ups I didn't test it, just looked at the code 🤡 

I thought that `menu-update.ts` needed to be called in order to update menu items... I'm gonna recheck the logic to get a better understanding of how it works :D

Reply:
Ah ok now I saw it! Both the `build-default-menu.ts` logic and the `menu-update.ts` logic gets executed when `refreshRepository()` is called.

The main difference that I see is that `menu-update.ts` is also called every time the AppStore emits a change (which I assume it happens way more often).

Have we considered having a single entrypoint to update the menu that only gets executed on `refreshRepository()`? Or there are scenarios where we want to update enabled value of an item where `refreshRepository()` is not called?

Reply:
> Have we considered having a single entrypoint to update the menu that only gets executed on `refreshRepository()`? Or there are scenarios where we want to update enabled value of an item where `refreshRepository()` is not called?

I think one scenario where we want to is when there's no repository selected. Now that you mention it though it seems odd that we update stuff in `refreshRepository`. I was under the impression that `refreshRepository` could be called for a repository which isn't currently selected  (i.e. when refreshing repository indicators).

As for the single entry point for both name and enabledness I think what happened is I wrote the stuff for toggling menu item enabledness and then the label stuff happened separately from that so chances are it's not a conscious decision but rather something that's evolved over time.